src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -611,7 +563,39 @@ object OrderedRVD {
         Array.tabulate(nPartitions - 1)(i => keys((i + 1) * k))
       }
 
-    UnsafeIndexedSeq(TArray(typ.pkType), rangeBounds)
+    val min = pkis.map(_.min).min(pkOrd)
+    val max = pkis.map(_.max).max(pkOrd)
+
+    OrderedRVDPartitioner.makeRangeBoundIntervals(typ.pkType, min +: partitionMaxes :+ max)
+  }
+
+  def shuffleAndAdjustBounds(typ: OrderedRVDType,
+    partitioner: OrderedRVDPartitioner,
+    rdd: RDD[RegionValue]): OrderedRVD = {
+    if (rdd.isEmpty())
```

Comment:
I don't prefer RDD.isEmpty.  Better to bail after getPartitionKeyInfo if it is empty.  Also, don't call shuffle when it is empty, just return OrderedRVD.empty.